lr-cli/cortx_setup/commands/node/finalize.py

```diff
@@ -35,86 +35,189 @@ class NodeFinalize(Command):
-        server_type = get_pillar_data(f'cluster/{minion_id}/type')
+        server_type = get_pillar_data(f'cluster/{node_id}/type')
         if not server_type or server_type is MISSED:
-            raise CortxSetupError(f"Server type is not provided")
+            self.logger.error("Server type is not provided")
         self.logger.debug("Server type check: Success")
         return server_type
 
+
+    def _create_field_users(self, node_id):
+        """users creation.
+
+        Creates field users - `nodeadmin` and `support`
+        with specific permissions each
+
+        """
+        user_passwd = "Seagate123!"
+#        server_hostname = get_pillar_data(f'cluster/{node_id}/hostname')
+        for user in ['nodeadmin', 'support']:
+            home_dir = Path(f'/home/{user}')
+#            ssh_dir = home_dir / '.ssh'
+
+            ugroup = user if user == "nodeadmin" else "root"
+
+            self.logger.debug(
+               f"Creating, if not present, a new user group '{ugroup}'"
+            )
+            StateFunExecuter.execute(
+                'group.present',
+                fun_kwargs=dict(
+                    name=ugroup
+                ),
+                targets=node_id
+            )
+
+            self.logger.debug(f"Creating sudoers file with permissions for user: '{user}'")
+            StateFunExecuter.execute(
+                'file.managed',
+                fun_kwargs=dict(
+                    name=f"/etc/sudoers.d/{user}",
+                    contents=(f'## Restricted access for {user} group users \n'
+                              f'%{user}   ALL = NOPASSWD: '
+                              '/opt/seagate/cortx/provisioner/node_cli/'),
+                    create=True,
+                    replace=True,
+                    user=ugroup,
+                    group=ugroup,
+                    mode=440
+                ),
+                targets=node_id
+            )
+
+            self.logger.debug(f"Creating user: '{user}'")
+            StateFunExecuter.execute(
+                'user.present',
+                fun_kwargs=dict(
+                    name=user,
+                    password=user_passwd,
+                    hash_password=True,
+                    home=str(home_dir),
+                    groups=ugroup,
+                    shell="/opt/seagate/cortx/provisioner/node_cli/"
+                ),
+                targets=node_id,
+                secure=True
+            )
+
+            self.logger.debug(f"Success: User creation for '{user}'")
+
+
     def run(self, force=False):
-        local_minion = local_minion_id()
-        self._validate_cert_installation()
-        self._validate_health_map()
-        self._validate_interfaces(local_minion)
-        s_type = self._validate_server_type(local_minion)
-        self._validate_devices(local_minion, s_type)
-        self.logger.debug("Done")
+        node_id = local_minion_id()
+        val_1 = self._validate_cert_installation()
```

Comment:
Lets use better names for variables:
val_1 change to validate_certificate
val_2 change to validate_healthmap
val_3 change to validate_interfaces
s_type change to server_type
val_4 change to validate_devices

OR

Instead use case as follows for each of the above checks (For example):
if self._validate_cert_installation():
    # Certificate validation success
elif force:
    # Certificate validation failed but skipped because of force
else:
    # Certificate validation failed (Throw exception)

Reply:
@nitin-seagate yes sure! this was initial draft to confirm if the flow is correct. Addressed now. Thanks

Reply:
This comment is not addressed

Reply:
Earlier, the final validations were written directly to `run` method and exception was thrown when an error occurs. 
Now, we're handling all the validations in `try`-`except` block, and if there's any error, it will check for `force` param and if not, only then will raise an exception to the user.